src/fragmentarium/ui/fragment/WordDownloadButton.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useState } from 'react'
+import { wordExport } from './WordExport'
+import { Fragment } from 'fragmentarium/domain/fragment'
+import WordService from 'dictionary/application/WordService'
+import { Dropdown } from 'react-bootstrap'
+import { saveAs } from 'file-saver'
+import Spinner from 'common/Spinner'
+import { Packer } from 'docx'
+
+type Props = {
+  children: React.ReactNode
+  fragment: Fragment
+  wordService: WordService
+}
+
+export default function WordDownloadButton({
+  fragment,
+  wordService,
+  children,
+}: Props): JSX.Element {
+  const [isLoading, setIsLoading] = useState(false)
+
+  const handleClick = (event) => {
+    setIsLoading(true)
+    getWordBlob(fragment, wordService).then((blob) => {
+      saveAs(blob, `${fragment.number}.docx`)
+      setIsLoading(false)
+    })
+  }
+
+  return (
+    <>
+      <Dropdown.Item onClick={handleClick}>
+        {isLoading ? <Spinner /> : children}
+      </Dropdown.Item>
+    </>
+  )
+}
+
+async function getWordBlob(
+  fragment: Fragment,
+  wordService: WordService
+): Promise<Blob> {
+  const wordDoc = await wordExport(fragment, wordService).then((doc) => {
+    return doc
+  })
+  const wordBlob: Blob = await Packer.toBlob(wordDoc).then((blob) => {
+    return blob
+  })
+  return wordBlob
```

Comment:
`then` with identity function is redundant. You should use Bluebird promises (see comment above). `await` will create a native JS promise which is not cancellable.

Reply:
removed identity function, is there an example in the code where a Bluebird promise is used?

Reply:
E.g. `BibliographyEntryForm`, `cancellableFetch`.

Reply:
now wrapped in Bluebird Promises